base/src/main/scala/co/uproot/abandon/Ast.scala

```diff
@@ -369,4 +375,12 @@ case class Scope(entries: Seq[ASTEntry], parentOpt: Option[Scope]) extends ASTEn
         includedScopes.foreach(_.checkDupes())
     }
   }
+
+  def checkUnusedSymbols() {
+    import Console.{YELLOW, BOLD, RESET}
+    definitions.filterNot(_.isUsed)
+      .foreach(d => {
+        println(s"${YELLOW}${BOLD}symbol '${d.name}' defined in ${d.pos.filename} line: ${d.pos.pos.line} but never used${RESET}")
+      })
+  }
```

Comment:
You have to make a recursive check for `includedScopes` as well. Test-case:

```
{
def y = 10
}
```

Note the braces which introduces a new scope.